src/moin/storage/middleware/indexing.py

```diff
@@ -148,6 +148,23 @@ def get_names(meta):
     return names
 
 
+def parent_names(names):
+    """
+    Compute list of parent names (same order as in names, but no dupes)
+
+    :param names: item NAME from whoosh index, where NAME is a list
+    :return: parent names list
+    """
+    parent_names = []
+    for name in names:
+        parentname_tail = name.rsplit('/', 1)
+        if len(parentname_tail) == 2:
+            parent_name = parentname_tail[0]
+            if parent_name not in parent_names:
+                parent_names.append(parent_name)
```

Comment:
is there some requirement for specific result order or for result being a list?

if not, you could also use a set, that would dedupe automatically.

Reply:
That code was just moved as was, will change to use set.